src/main.rs

```diff
@@ -1,3 +1,103 @@
+use sha3::{Digest, Keccak256};
+
+type MerkleTree = Vec<Vec<String>>;
+
+/// Checks if the given hash is present at any level of the Merkle Tree
+fn contains_hash(merkle: &MerkleTree, hash: &String) -> bool {
+    merkle.iter().any(|level| level.contains(hash))
+}
+
+/// Creates the next (upper) level for a Merkle Tree given the previous level.
+fn calculate_next_level(vec: &[String]) -> Vec<String> {
```

Comment:
Maybe the name of the input `vec` can be changed to something more meanigful like `previous_level`.